websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,159 @@
+const presence = new Presence({
+		clientId: "580032576434077707",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video = {
+	current: 0,
+	duration: 0,
+	paused: true,
+};
+
+enum image {
+	search = "https://i.imgur.com/1VwDv4D.png",
+	home = "https://i.imgur.com/xVhTZcB.png",
+	play = "https://i.imgur.com/Zd2WTR1.png",
+	pause = "https://i.imgur.com/LV9xIjD.png",
+	logo = "https://i.imgur.com/KEkxfKO.png",
+}
+
+presence.on(
+	"iFrameData",
+	(data: { current: number; duration: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: image.logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, search } = document.location;
+	presenceData.smallImageKey = image.search;
+	if (search !== "") {
+		presenceData.details = "Używa zaawansowanej wyszukiwarki...";
+		const searchvalues = search.split("&");
+		for (const searchvalue of searchvalues) {
+			if (searchvalue.startsWith("s=")) {
+				presenceData.details = "Wyszukuje po frazie:";
+				presenceData.state = searchvalue
+					.substring(2, searchvalue.length)
+					.replaceAll("+", " ")
+					.replaceAll("%20", " ");
```

Comment:
Depends on the context. If you want e.g. `Example%20+test` to be `Example test` then use this:
```suggestion
					.replace(/(%20)(\+)/gm, " ")
```

Otherwise If you want e.g. `Example%20+test` to be `Example  test` then use this:
```suggestion
					.replace(/(%20)|(\+)/gm, " ")
```

Reply:
If in search query is only ?s=x, spaces are `+`, but if there is only type in search query like genre/studio etc, spaces are `%20`, so which one i should use?

Reply:
> If in search query is only ?s=x, spaces are `+`, but if there is only type in search query like genre/studio etc, spaces are `%20`, so which one i should use?

this one:
```suggestion
					.replace(/(%20)(\+)/gm, " ")
```